### ISSUE TITLE
I developing a BigQuery "driver" to GORM v2 and looking to share/contribute/suggestions.

### ISSUE THREAD

**Issue by guypeled76** [go-gorm/gorm#3138]
## Your Question

I am developing a GORM V2 "driver" to BigQuery as I was frustrated with the currently tools to work with bigquery queries. 
It's real fresh out of the oven and I am now starting to work with it in a project I am working on.
I will continue to work on it on as my project goes forward and I will find bugs / more features I need.

You can see the go code here:
https://github.com/guypeled76/go-bigquery-driver 

## Expected answer

Opinions and suggestions both for the code itself and for sharing it.

Thanks,
Guy

**Comment by jinzhu** [comment]
Thank you for your contribution! 

Let me know if you would like to have a go-gorm repo for hosting the project ;)

**Comment by jinzhu** [comment]
Or if you find anything missing in GORM that can't fully support BigQuery, just send me a feature request.

**Comment by guypeled76** [comment]
Thank you very much.
I would love to have a go-gorm repo for hosting the project so people will use it and it gets better.
I do have one thing that might be nice to have within the core of GORM and not implemented as I did.

BigQuery has less use for actual joining tables although it is done from time to time. It is more common to find records and arrays of records nested. From my perspective both could be considered nested tables which means a column that returns a table or in sql driver terms a "sql.Rows" go interface implementation.  

Currently I have implemented an adapter concept where I use the new context based query to pass down an adapter that adapts the data at the sql driver level to fit what GORM expects which is go structs and arrays/slices. I kind of like create a fake sql query that returns the nested table and I scan it into the target struct/slice. It would be nice if GORM would allow some king of flag like "NESTED" (borrowed from "EMBEDDED" which will let GORM except a  column to have "sql.Rows" in it so it can scan to the target struct/slice. 

That said the current implementation wraps the "sql.Rows" with a sql driver rows implementation. It might be hard to do that.
It's not super important as I got around it and if in the future I will find out that it is limiting I might go for a GORM PR on the issue. I predict it will be a very small change to the code.


Thanks,
Guy
